### PR TITLE
Fixes #2588: add helper to display toasts

### LINKED ISSUE #2588: Add a helper to display Toast in dashboard
**What would you like to be added**:
A helper to display [toasts](https://getbootstrap.com/docs/5.0/components/toasts)
```ts
// Display a toast on screen for a fixed X seconds
const enqueueToast = (message: string, variant: "error" | "success" | "info" | "warning"): void => {

}
```

**Why would you like this to be added**:
I'm working on #2587 and I need a way to inform users if the content is successfully copied to the clipboard or something wrong has occurred. Also to replace the `console.error(error)` in catch.

### PR DESCRIPTION
**Pull request by Andrewnt219**
## Issue This PR Addresses
Closes #2588 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

- `toast.js` exposes a single method `showToast` to show different kinds of notifications to users.
- HTML and JS for toast are added globally to all the pages.

https://user-images.githubusercontent.com/52666982/145652785-6219434d-0b4e-4b9c-91e2-313382935de7.mp4

## Test plan
- Go to src/api/status/public/js/toast.js
- At the end of the file, invoke as many `enqueueToast` as you like
```js
// toasts.js

showToast("Hello world", "info");
showToast("Help!", "danger");
```

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — APPROVED** [review]

**Review by DukeManh — APPROVED** [review]

File: src/api/status/public/js/utils/toast.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * @returns {HTMLElement} the container for all stacking toast elements
+ */
+const queryToastContainer = () => document.getElementById('toast-container');
+
+/**
+ * Make a new empty toast node from the sample toast node
+ *
+ * @returns {HTMLElement} an empty toast element
+ */
+const cloneDummyToastNode = () => {
```
Comment (humphd) [inline src/api/status/public/js/utils/toast.js:20]: You use Element and Node in this file, and I wonder if you should standardize.  An Element becomes a node in the DOM when inserted, but I think here you are really returning an Element.

File: src/api/status/public/js/utils/toast.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+/**
+ * Create a toast element and modify the content
+ *
+ * @param {string} message to display
+ * @param {ToastVariant} variant for toast UI theme
+ * @returns {HTMLElement} styled toast element with custom message
+ */
+const createToastNode = (message, variant) => {
```
Comment (humphd) [inline src/api/status/public/js/utils/toast.js:40]: Is it worth defining a default value here (info?)

File: src/api/status/public/js/utils/toast.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Show a toast with custom message to users
+ *
+ * @param {string} message to display
+ * @param {ToastVariant} variant for toast UI theme
+ * @param {any} [options] bootstrap toast options. See {@link https://getbootstrap.com/docs/5.0/components/toasts/#options}
+ * @returns {any} an instance of BootStrap Toast. See {@link https://getbootstrap.com/docs/5.0/components/toasts/#methods}
+ */
+const enqueueToast = (message, variant, options) => {
```
Comment (humphd) [inline src/api/status/public/js/utils/toast.js:60]: I don't love this name.  It's correct, but I'd suggest `showToast` is what callers will be thinking about.

File: src/api/status/public/js/utils/toast.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/**
+ * UI theme for toast
+ * @typedef {("danger" | "success" | "info" | "warning" | "primary")} ToastVariant
```
Comment (humphd) [inline src/api/status/public/js/utils/toast.js:3]: Really nice to see TS in JS, we should do more of this throughout Telescope.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/status/public/js/utils/toast.js (hunk 1 above)
Comment (Andrewnt219) [reply src/api/status/public/js/utils/toast.js:20]: You're right, I thought node and element are the same. When I read on MDN again, `Element` is a subclass of `Node`, and the `toast` is indeed an Element.

File: src/api/status/public/js/utils/toast.js (hunk 2 above)
Comment (Andrewnt219) [reply src/api/status/public/js/utils/toast.js:40]: Let's do that.

File: src/api/status/public/js/utils/toast.js (hunk 4 above)
Comment (Andrewnt219) [reply src/api/status/public/js/utils/toast.js:3]: After working with the TypeScript codebase, I don't want any soul to go through the same thing again.

Comment (Andrewnt219) [reply src/api/status/public/js/utils/toast.js:3]: It also adds a nice hint to the users.

image

**Review by humphd — APPROVED** [review]

**Review by DukeManh — APPROVED** [review]

**Pull request merged**